charms/kfp-api/src/charm.py

```diff
@@ -210,64 +207,63 @@ def _kfp_api_layer(self) -> Layer:
             raise error
 
-        # at this point all data is correctly populated and proper config can be generated
-        config_json = {
-            "DBConfig": {
-                "ConMaxLifeTime": "120s",
-                "DBName": db_data["db_name"],
-                "DriverName": "mysql",
-                "GroupConcatMaxLen": "4194304",
-                "Host": db_data["db_host"],
-                "Password": db_data["db_password"],
-                "Port": db_data["db_port"],
-                "User": db_data["db_username"],
-            },
-            "ObjectStoreConfig": {
-                "AccessKey": os["access-key"],
-                "BucketName": config["object-store-bucket-name"],
-                "Host": f"{os['service']}.{os['namespace']}",
-                "Multipart": {"Disable": "true"},
-                "PipelinePath": "pipelines",
-                "Port": str(os["port"]),
-                "Region": "",
-                "SecretAccessKey": os["secret-key"],
-                "Secure": str(os["secure"]).lower(),
-            },
-            "ARCHIVE_CONFIG_LOG_FILE_NAME": config["log-archive-filename"],
-            "ARCHIVE_CONFIG_LOG_PATH_PREFIX": config["log-archive-prefix"],
-            "AUTO_UPDATE_PIPELINE_DEFAULT_VERSION": str(
-                config["auto-update-default-version"]
-            ).lower(),
-            "CACHE_IMAGE": config["cache-image"],
-            "CACHE_NODE_RESTRICTIONS": "false",
-            "CacheEnabled": str(config["cache-enabled"]).lower(),
-            "DefaultPipelineRunnerServiceAccount": config["runner-sa"],
-            "InitConnectionTimeout": config["init-connection-timeout"],
+        env_vars = {
+            "AUTO_UPDATE_PIPELINE_DEFAULT_VERSION": self.model.config[
+                "auto-update-default-version"
+            ],
+            "KFP_API_SERVICE_NAME": "apiserver",
             "KUBEFLOW_USERID_HEADER": "kubeflow-userid",
             "KUBEFLOW_USERID_PREFIX": "",
+            "POD_NAMESPACE": self.model.name,
+            "OBJECTSTORECONFIG_SECURE": "false",
+            "OBJECTSTORECONFIG_BUCKETNAME": self.model.config[
+                "object-store-bucket-name"
+            ],  # mlpipeline
+            "DBCONFIG_USER": db_data["db_username"],  # root
+            "DBCONFIG_PASSWORD": db_data["db_password"],
+            "DBCONFIG_DBNAME": db_data["db_name"],
+            "DBCONFIG_HOST": db_data["db_host"],
+            "DBCONFIG_PORT": db_data["db_port"],
+            "DBCONFIG_CONMAXLIFETIME": "120s",  # 120s
+            "DB_DRIVER_NAME": "mysql",  # mysql
+            "DBCONFIG_MYSQLCONFIG_USER": db_data["db_username"],  # root
+            "DBCONFIG_MYSQLCONFIG_PASSWORD": db_data["db_password"],
+            "DBCONFIG_MYSQLCONFIG_DBNAME": db_data["db_name"],  # mlpipeline
+            "DBCONFIG_MYSQLCONFIG_HOST": db_data["db_host"],  # mysql
+            "DBCONFIG_MYSQLCONFIG_PORT": db_data["db_port"],  # 3306
```

Comment:
Regarding comments with expected values, do we need them here? Are they here by accident?

Reply:
I left them behind as a reference, do you think they are misleading?